### PR TITLE
fix: FieldMetadata default value and options better validation

### LINKED ISSUE #2754: Set, retrieve and update default `currencyCode` for Currency Field
## Scope & Context

See https://github.com/twentyhq/twenty/issues/2330

When creating a Currency field, the user selects a default currency code and can later view and edit the default currency code.

## Current behavior

- a default value can be set on field creation, but `amountMicros` is not nullable
- the field metadata default value cannot be retrieved or updated

## Expected behavior

We need to be able to:
- set a default `currencyCode` without having to set a default `amountMicros`
- retrieve and update the default `currencyCode` so it can be viewed and edited in the Field Edition form in Settings

### PR DESCRIPTION
**Pull request by magrinj**
This PR, in how `defaultValue` and `options` were validated.
It nos use asynchronous validators to support `create` and `update` validation based on the type of the field.

### CODE REVIEW

File: server/patches/class-validator+0.14.0.patch
Comment (magrinj) [inline server/patches/class-validator+0.14.0.patch:1]: Can't generate something smaller for now... As it's map file and bundled sources

File: server/src/metadata/field-metadata/validators/is-field-metadata-options.validator.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      const fieldMetadata = await this.fieldMetadataService.findOneOrFail(id);
+
+      type = fieldMetadata.type;
+    }
+
+    return validateOptionsForType(type, value);
+  }
+
+  defaultMessage(): string {
+    return 'FieldMetadataDefaultValue is not valid';
```
Comment (Weiko) [inline server/src/metadata/field-metadata/validators/is-field-metadata-options.validator.ts:40]:
```suggestion
    return 'FieldMetadataOptions is not valid';
```

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "configurations": [
```
Comment (Weiko) [inline .vscode/launch.json:2]: Is it intended to push this?

**Review by Weiko — COMMENTED** [review]
Left a few comments, LGTM 👏

File: .vscode/launch.json (hunk 2 above)
Comment (magrinj) [reply .vscode/launch.json:2]: Yes, just to launch the NestJS project in debug mode, with breakpoints and all

File: server/src/metadata/field-metadata/field-metadata.service.ts [hunk 3]
```diff
@@ -207,6 +207,25 @@ export class FieldMetadataService extends TypeOrmQueryService<FieldMetadataEntit
     return updatedFieldMetadata;
   }
 
+  public async findOneOrFail(
+    id: string,
+    options?: FindOneOptions<FieldMetadataEntity>,
+  ) {
+    const fieldMetadata = await this.fieldMetadataRepository.findOne({
```
[the remark this answers is not in the document]
Comment (magrinj) [reply server/src/metadata/field-metadata/field-metadata.service.ts:214]: Are you sure that an exception is thrown if the object doesn't exist ?
The type of the return can be `null`

File: server/src/metadata/field-metadata/field-metadata.resolver.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      workspaceId,
+    });
+  }
+
+  @Mutation(() => FieldMetadataDTO)
+  updateOneField(
+    @Args('input') input: UpdateOneFieldMetadataInput,
+    @AuthWorkspace() { id: workspaceId }: Workspace,
+  ) {
+    return this.fieldMetadataService.updateOne(input.id, {
```
[the remark this answers is not in the document]
Comment (magrinj) [reply server/src/metadata/field-metadata/field-metadata.resolver.ts:33]: @Weiko I'll take a look on that

[the remark this answers is not in the document]
Comment (magrinj) [reply server/src/metadata/field-metadata/field-metadata.resolver.ts:33]: You're right, I've fix the issue

**Review by Weiko — APPROVED** [review]
LGTM!

File: server/src/metadata/field-metadata/field-metadata.service.ts [hunk 5]
```diff
@@ -159,6 +159,15 @@ export class FieldMetadataService extends TypeOrmQueryService<FieldMetadataEntit
       throw new NotFoundException('Field does not exist');
     }
 
+    if (existingFieldMetadata.isCustom === false) {
+      // We can only update the isActive field for standard fields
```
[the remark this answers is not in the document]
Comment (magrinj) [reply server/src/metadata/field-metadata/field-metadata.service.ts:163]: @Weiko Yes it's a quick change, we can maybe create another default validator for `isCustom` properties, but you told me that the front-end send all the input informations and I didn't want to break it

**Pull request merged**